mllib/src/main/scala/org/apache/spark/ml/feature/StandardScaler.scala

```diff
@@ -48,6 +48,16 @@ private[feature] trait StandardScalerParams extends Params with HasInputCol with
    * @group param
    */
   val withStd: BooleanParam = new BooleanParam(this, "withStd", "Scale to unit standard deviation")
+
+  protected def validateAndTransformSchema(schema: StructType): StructType = {
+    val inputType = schema($(inputCol)).dataType
```

Comment:
This could be further simplified by using the SchemaUtils from `ml.util`. See:
https://github.com/apache/spark/blob/836a75898fdc4b10d4d00676ef29e24cc96f09fd/mllib/src/main/scala/org/apache/spark/ml/feature/IDF.scala#L55